server/sonar-server/src/main/java/org/sonar/server/ce/ws/TaskFormatter.java

```diff
@@ -69,29 +71,21 @@ public TaskFormatter(DbClient dbClient, System2 system2) {
     WsCe.Task.Builder builder = WsCe.Task.newBuilder();
     String organizationKey = componentDtoCache.getOrganizationKey(dto.getComponentUuid());
     // FIXME organization field should be set from the CeQueueDto rather than from the ComponentDto
-    if (organizationKey != null) {
-      builder.setOrganization(organizationKey);
+    setNullable(organizationKey, builder::setOrganization);
+    if (dto.getComponentUuid() != null) {
+      builder.setComponentId(dto.getComponentUuid());
+      buildComponent(builder, componentDtoCache.getComponent(dto.getComponentUuid()));
     }
     builder.setId(dto.getUuid());
     builder.setStatus(WsCe.TaskStatus.valueOf(dto.getStatus().name()));
     builder.setType(dto.getTaskType());
     builder.setLogs(false);
-    if (dto.getComponentUuid() != null) {
-      builder.setComponentId(dto.getComponentUuid());
-      buildComponent(builder, componentDtoCache.getComponent(dto.getComponentUuid()));
-    }
-    if (dto.getSubmitterLogin() != null) {
-      builder.setSubmitterLogin(dto.getSubmitterLogin());
-    }
-    builder.setSubmittedAt(DateUtils.formatDateTime(new Date(dto.getCreatedAt())));
+    setNullable(dto.getSubmitterLogin(), builder::setSubmitterLogin);
+    builder.setSubmittedAt(formatDateTime(new Date(dto.getCreatedAt())));
     if (dto.getStartedAt() != null) {
```

Comment:
why don't use use setNullable here too ?
Sure, you won't be able to use a method reference, but you can still write a lamba